src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -1350,41 +1359,77 @@ private static void WriteSpaceForHash(Stream peStream, int strongNameSignatureSi
             }
         }
 
+        /// <summary>
+        /// Write one entry in the "Debug Directory (Image Only)"
+        /// See https://msdn.microsoft.com/en-us/windows/hardware/gg463119.aspx
+        /// section 5.1.1 (pages 71-72).
+        /// </summary>
+        private static void WriteDebugTableEntry(
+            PooledBlobBuilder writer,
+            byte[] stamp,
+            uint version, // major and minor version, combined
+            uint debugType,
+            uint sizeOfData,
+            uint addressOfRawData,
+            uint pointerToRawData
+            )
+        {
+            writer.WriteUInt32(0); // characteristics
+            Debug.Assert(stamp.Length == 4);
+            writer.WriteBytes(stamp);
+            writer.WriteUInt32(version);
+            writer.WriteUInt32(debugType);
+            writer.WriteUInt32(sizeOfData);
+            writer.WriteUInt32(addressOfRawData);
+            writer.WriteUInt32(pointerToRawData);
+        }
+
+        private static byte[] zeroStamp = new byte[4]; // four bytes of zero
+
+        /// <summary>
+        /// Write the entire "Debug Directory (Image Only)" along with data that it points to.
+        /// </summary>
+        /// <param name="peStream"></param>
+        /// <param name="textSection"></param>
+        /// <param name="nativePdbContentId"></param>
+        /// <param name="portablePdbContentId"></param>
+        /// <param name="metadataSizes"></param>
         private void WriteDebugTable(Stream peStream, SectionHeader textSection, ContentId nativePdbContentId, ContentId portablePdbContentId, MetadataSizes metadataSizes)
         {
+            int tableSize = ImageDebugDirectoryBaseSize;
             Debug.Assert(nativePdbContentId.IsDefault ^ portablePdbContentId.IsDefault);
 
             var writer = PooledBlobBuilder.GetInstance();
 
-            // characteristics:
-            writer.WriteUInt32(0);
-
-            // PDB stamp & version
-            if (portablePdbContentId.IsDefault)
-            {
-                writer.WriteBytes(nativePdbContentId.Stamp);
-                writer.WriteUInt32(0);
-            }
-            else
+            const int ImageDebugTypeCodeView = 2;
```

Comment:
Sorry for not being clear. Names of locals shouldn't start with capital letters.